### PR TITLE
Disable Sentry with SentryOptions

### LINKED ISSUE #2833: Not possible to pass empty string into DSN under logback
### Integration

sentry-logback

### Java Version

20

### Version

6.25.0

### Steps to Reproduce

Sentry.java shows an exception `throw new IllegalArgumentException("DSN is required. Use empty string to disable SDK.");`

However, it is impossible to pass an empty string as the DSN in logback.xml because Logback interprets an empty string value as the property not existing altogether.

The only thing that seems to work is to pass an undefined property as the value because `SentryAppender` checks if the DSN ends with "IS_UNDEFINED".

### Expected Result

Either document a way to pass an empty string into dsn in logback.xml, or update the documentation and exception message to indicate what we *are* supposed to do under Logback.

### Actual Result

N/A

### PR DESCRIPTION
**Pull request by lbloder**
## :scroll: Description
Adds `enabled` flag to `SentryOptions` and `ExternalOptions` to allow users to disable Sentry.

## :bulb: Motivation and Context
Fixes #2833 

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Features
+
+- Disable Sentry with SentryOptions ([#2840](https://github.com/getsentry/sentry-java/pull/2840))
```
Comment (adinauer) [inline CHANGELOG.md:7]:
```suggestion
- You can now disable Sentry using a new `enabled` option ([#2840](https://github.com/getsentry/sentry-java/pull/2840))
```

File: sentry/src/main/java/io/sentry/Sentry.java [hunk 2]
```diff
@@ -288,8 +288,9 @@ private static boolean initConfigurations(final @NotNull SentryOptions options)
 
     final String dsn = options.getDsn();
     if (dsn == null) {
-      throw new IllegalArgumentException("DSN is required. Use empty string to disable SDK.");
-    } else if (dsn.isEmpty()) {
+      throw new IllegalArgumentException(
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/Sentry.java:291]: We should check the `enabled` flag first and not throw if disabled regardless of DSN.

Comment (lbloder) [reply sentry/src/main/java/io/sentry/Sentry.java:291]: Yeah, makes sense, any preference between the following two variants?

```
    if(!options.isEnabled() || (dsn != null && dsn.isEmpty())) {
      close();
      return false;
    } else if (dsn == null) {
      throw new IllegalArgumentException(
        "DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.");
    }
``` 

```
    if (!options.isEnabled()) {
      close();
      return false;
    }

    if (dsn == null) {
      throw new IllegalArgumentException(
          "DSN is required. Use empty string or set enabled to false in SentryOptions to disable SDK.");
    } else if (dsn.isEmpty()) {
      close();
      return false;
    }
```

Comment (adinauer) [reply sentry/src/main/java/io/sentry/Sentry.java:291]: Top one seems a bit better as it doesn't duplicate the code.

**Review by adinauer — APPROVED** [review]
LGTM

**Review by romtsn — APPROVED** [review]
LGTM, missing Manifest option, but we're gonna drop it in the next major most likely (v8) and this is probably anyway not gonna be used there, so it's alright

**Comment by benoit-lateltin** [comment]
Nice PR! Would it be possible to have this `sentry.enabled` property hot-reloadable via a Spring Cloud Config Server ? Could we use the [Environment](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/env/Environment.html) PropertyResolver to retrieve the prop when processing the event?

**Comment by adinauer** [comment]
Hello @benoit-lateltin I've created https://github.com/getsentry/sentry-java/issues/2939 to track your feature request. Thanks for the suggestions - makes sense. Can't say when we'll get to implementing it though.

**Pull request merged**